tests/chainer_tests/functions_tests/array_tests/test_squeeze.py

```diff
@@ -27,50 +43,28 @@
         {'dtype': numpy.float64},
     ],
 ))
-class TestSqueeze(unittest.TestCase):
+class TestSqueeze(testing.FunctionTestCase):
 
     def setUp(self):
-        self.x = numpy.random.uniform(-1, 1, (1, 1, 3, 1)).astype(self.dtype)
-        self.g = numpy.random.uniform(-1, 1, self.out_shape).astype(self.dtype)
-
         self.check_forward_options = {}
-        self.check_backward_options = {'dtype': numpy.float64}
+        self.check_backward_options = {}
         if self.dtype == numpy.float16:
             self.check_forward_options = {'atol': 5e-4, 'rtol': 5e-3}
-            self.check_backward_options = {
-                'dtype': numpy.float64, 'atol': 2 ** -4, 'rtol': 2 ** -4}
-
-    def check_forward(self, x_data):
-        y = functions.squeeze(x_data, axis=self.axis)
-        expected = numpy.squeeze(self.x, axis=self.axis)
-        testing.assert_allclose(y.data, expected, **self.check_forward_options)
+            self.check_backward_options.update({
+                'atol': 2 ** -4, 'rtol': 2 ** -4})
```

Comment:
Do not mix styles (`options = {...}` and `options.update({...})`).  I'd always use `update` because `self.check_forward_options`, `self.check_backward_options`, and `self.check_double_backward_options` are `{}` by default.

BTW, does the test require large tolerances for float16?  Did you check double-backward tests pass without adjusting tolerances?

Reply:
Thanks.. @toslunar . Made the required changes. I'll keep your words in mind. 

Also, I did try adjusting the relative and absolute tolerances after your comment and they turn out to be extremely low. Should I update the `atol` and `rtol` as well?

Reply:
It's no problem to leave too large tolerances.  Could you set `self.check_double_backward_options` if you found the double-backward tests are possible to fail.  Double backward is often (but not always) less accurate than backward.

Reply:
@toslunar  I find that double backward tests are likely to fail when tolerances are less than the order of 1e-13 which is again extremely low. Should I update the `self.check_double_backward_options`. If yes then what should be the order of magnitude for absolute and relative tolerance values?

Also is there any specific paradigm to follow while setting up or calculating tolerances?

Reply:
No.



> Should I update the `self.check_double_backward_options`.

No.

> Also is there any specific paradigm to follow while setting up or calculating tolerances?

`chainer.testing.condition.repeat` is often used to confirm stability of tests.

Reply:
@toslunar Thanks. I think the commit is ready to be merged then.